src/ImageProcessor.Web/HttpModules/ImageProcessingModule.cs

```diff
@@ -826,7 +826,10 @@ private IImageService GetImageServiceForRequest(string url, string applicationPa
             foreach (IImageService service in services)
             {
                 string key = service.Prefix;
-                if (!string.IsNullOrWhiteSpace(key) && path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrWhiteSpace(key) 
+                    && (service.IsRegex 
```

Comment:
But wouldn't escaping the prefix mean that you cannot actually use a regular expression as the prefix? Effectively converts special characters in the regex to a literal characters?
For example, an ImageService with regular expression for the prefix `prefix="firstprefix|secondprefix"` to match these urls:
`http://www.domain.com/firstprefix/image.jpg`
`http://www.domain.com/secondprefix/image.jpg`

Regex escaping this prefix will give `firstprefix\|secondprefix`

Both sides of this OR expression would be false?
```(Regex.IsMatch(path, Regex.Escape(key), RegexOptions.IgnoreCase) 
        || path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase)```